### PR TITLE
[CI] Disable Ray log deduplication for doctests

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

**tl;dr** This PR fixes `:book: Documentation`

https://github.com/ray-project/ray/pull/33403 introduced Ray log deduplication. This feature causes `testcode` outputs to get buffered and printed in later `testcode`s.

> However, subsequent log messages of the same pattern (ignoring words with numeric components) will be buffered for up to five seconds and printed in batch.

To prevent non-deterministic doctests, this PR disables the feature entirely in CI.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — APPROVED** [review]

**Review by ericl — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging into master
